### PR TITLE
Add shortform "r" and "w" capability names to pgen

### PR DESCRIPTION
**Pull request by geky**
This makes it easier to write policy files since read/write are probably the most common capability sets.

``` bash
$ pgen --capability input:rw,output:r
```

The exact mapping is taken from the test-collateral Makefile which has similar defines internally:

```
r = FD_READ + FD_SEEK + PATH_OPEN
w = FD_WRITE + PATH_CREATE_FILE + PATH_FILESTAT_SET_SIZE + FD_SEEK + PATH_OPEN
```

Note this only affects the `pgen` CLI, the policy file is still limited to the low-level WASI rights bitmask.

I put this together to move https://github.com/veracruz-project/veracruz/pull/75 along, welcome to feedback if we want to handle this differently. I can switch back to bitmasks if that's what we want to go with, it just requires magic constants on the command line.

---

Another option would be to add a more complicated/complete grammar, maybe something like:

``` bash
$ pgen --capability input:path_open+fd_read,output:path_open+fd_write+fd_seek
```

or

``` bash
$ pgen --capability input:path_open+fd_read,output:rw-fd_seek
```

Currently depending on this in https://github.com/veracruz-project/veracruz/pull/75

### CODE REVIEW

**Comment by geky** [comment]
CC @dominic-mulligan-arm, @ShaleXIONG

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 793da94a79dd023f773d9b43a978997a91e0863f
* Result: SUCCEEDED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=a6bab74a-b833-4a2f-a060-720508a49ecf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by ShaleXIONG — DISMISSED** [review]

File: test-collateral/generate-policy/src/main.rs [hunk 1]
```diff
@@ -220,7 +220,7 @@ fn parse_command_line() -> Arguments {
                 .short("p")
                 .long("capability")
                 .value_name("CAPABILITIES")
-                .help("The capability table of a client or a program of the form 'output:rw,input-0:w,prorgam.wasm:x' where each entry is separated by ','")
+                .help("The capability table of a client or a program of the form 'output:rw,input-0:w,prorgam.wasm:w' where each entry is separated by ','. These may be either some combination of 'r' and 'w' for reading and writing permissions respectively, or an integer containing the bitwise-or of the low-level WASI capabilities.")
```
Comment (dominic-mulligan-arm) [inline test-collateral/generate-policy/src/main.rs:223]: There's a spelling error here that was present before your change, but we may as well fix it here: "prorgam" --> "program".

File: test-collateral/generate-policy/src/main.rs [hunk 2]
```diff
@@ -16,6 +16,7 @@ use std::{
     path::{Path, PathBuf},
     process::{exit, Command},
     str::FromStr,
+    convert::TryFrom,
```
Comment (dominic-mulligan-arm) [inline test-collateral/generate-policy/src/main.rs:19]: Can you keep the imports in alphabetical order?

File: test-collateral/generate-policy/src/main.rs [hunk 3]
```diff
@@ -630,19 +630,51 @@ fn serialize_capability(cap_string : &[String]) -> Vec<FileRights> {
...
-    let number = string_number
-                .trim()
-                .parse::<u32>()
-                .expect(&format!("Failed to parse {}, not a u64", string_number));
-    // check if this is a valid number
-    let _cap = Rights::from_bits(number as u64)
-                .expect(&format!("Failed to parse {}, not a u64 representing WASI Right", number));
-    FileRights::new(file_name.trim().to_string(),number)
+        .next()
+        .expect(&format!("Failed to parse `{}`, contain no `:`", cap_string))
```
Comment (dominic-mulligan-arm) [inline test-collateral/generate-policy/src/main.rs:656]: sp: "contain" --> "contains"

File: test-collateral/generate-policy/src/main.rs (hunk 2 above)
Comment (geky) [reply test-collateral/generate-policy/src/main.rs:19]: I need to run rust fmt, I'll manually reorder these if that doesn't

File: test-collateral/generate-policy/src/main.rs (hunk 1 above)
Comment (geky) [reply test-collateral/generate-policy/src/main.rs:223]: Can do

File: test-collateral/generate-policy/src/main.rs (hunk 3 above)
Comment (geky) [reply test-collateral/generate-policy/src/main.rs:656]: Can do (not my fault... probably...)

**Comment by dominic-mulligan-arm** [comment]
@geky can you rebase this?  It looks like this PR is next in line for merging...

**Comment by dominic-mulligan-arm** [comment]
@ShaleXIONG have you adopted this PR?  Can you rebase it onto `main` ready for review?

**Review by dominic-mulligan-arm — APPROVED** [review]
LGTM.

**Comment by dominic-mulligan-arm** [comment]
Can this be rebased so we can merge it next?

**Review by dominic-mulligan-arm — APPROVED** [review]

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 57f5d51c8ed0499e0eccc83fadd076fe8af8d784
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=bb0a9071-ee00-41fb-bafe-9e05e3b2ee9a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by dominic-mulligan-arm** [comment]
+1 reached, merging.

**Pull request merged**